src/materials/DamagePlasticityStressUpdate.C

```diff
@@ -336,37 +325,27 @@ DamagePlasticityStressUpdate::dflowPotential_dstress(
                           .dsecondInvariant();
   RankTwoTensor d_sqrt_2J2;
   RankFourTensor dfp;
-  Real pre;
+  RankTwoTensor d2J2_dsigi_dsigj =
+      RankTwoTensor(2. / 3., 2. / 3., 2. / 3., -1. / 3., -1. / 3., -1. / 3.);
+  std::vector<Real> dJ2_dsigi(3);
+  for (unsigned i = 0; i < 3; ++i)
+    dJ2_dsigi[i] =
+        (2 * stress_params[i] - stress_params[(i + 1) % 3] - stress_params[(i + 2) % 3]) / 3;
+
   if (J2 == 0)
   {
-    d_sqrt_2J2 = RankTwoTensor();
-    dfp = RankFourTensor();
-    pre = 0;
+    for (unsigned i = 0; i < 3; ++i)
+      for (unsigned j = 0; j < 3; ++j)
+        dr_dstress[i][j] = 0.0;
   }
   else
   {
-    d_sqrt_2J2 = 0.5 * std::sqrt(2.0 / J2) * dII;
-    dfp = 0.5 * std::sqrt(2.0 / J2) *
-          RankTwoTensor(stress_params[0], stress_params[1], stress_params[2], 0, 0, 0)
-              .d2secondInvariant();
-    pre = -0.25 * std::sqrt(2.0) * std::pow(J2, -1.5);
+    for (unsigned i = 0; i < 3; ++i)
+      for (unsigned j = 0; j < 3; ++j)
+        dr_dstress[i][j] =
+            0.5 * (std::sqrt(2.0 / J2) * d2J2_dsigi_dsigj(i, j) -
+                   (1 / std::sqrt(2)) * std::pow(J2, -1.5) * dJ2_dsigi[i] * dJ2_dsigi[j]);
```

Comment:
I'm kind of curious -- I suspect that 
```
1.0/std::sqrt(Utility::pow<3>(J2))
```
is faster than:
```
std::pow(J2, -1.5)
```
I guess we'd need to test it, though. You can leave this as-is, but if you want to investigate, please do.

Reply:
left as is

Reply:
I just checked, and the suggestion that uses Utility::pow is a lot faster.